src/scrot.c

```diff
@@ -400,38 +400,54 @@ Window scrotGetWindow(Display *display, Window window, int x, int y)
 void scrotGrabMousePointer(const Imlib_Image image, const int xOffset,
     const int yOffset)
 {
-    XFixesCursorImage *xcim = XFixesGetCursorImage(disp);
-
-    if (!xcim) {
-        warnx("Failed to get mouse cursor image.");
-        return;
+    XFixesCursorImage *xcim = NULL;
+    char *errmsg = NULL;
+    unsigned short width;
+    unsigned short height;
+    int x;
+    int y;
+    size_t pixcnt;
+    uint32_t *pixels = NULL;
+    Imlib_Image imcursor;
+
+    /* Convert an X cursor into the format imlib wants. */
+    if ((xcim = XFixesGetCursorImage(disp)) == NULL) {
+        errmsg = "Can't get the cursor from X";
+        goto end;
     }
-
-    const unsigned short width = xcim->width;
-    const unsigned short height = xcim->height;
-    const int x = (xcim->x - xcim->xhot) - xOffset;
-    const int y = (xcim->y - xcim->yhot) - yOffset;
-    DATA32 data[width * height];
-    
-    for (size_t i = 0; i < width*height; i++)
-            data[i] = xcim->pixels[i];
-
-    Imlib_Image imcursor = imlib_create_image_using_data(width, height, data);
-
-    XFree(xcim);
-
+    width = xcim->width;
+    height = xcim->height;
+    x = (xcim->x - xcim->xhot) - xOffset;
+    y = (xcim->y - xcim->yhot) - yOffset;
+    pixcnt = width*height;
```

Comment:
Since `width` and `height` are `unsigned short`, the multiplication takes place in `int` after integer promotion. Cast one of the operands to `size_t` so that the mul happens at `size_t`.

```suggestion
    pixcnt = (size_t)width*height;
```

Reply:
Sure.